// @chirpstack/chirpstack-api/api/device_profile_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var api_device_profile_pb = require('../api/device_profile_pb.js');
var common_common_pb = require('../common/common_pb.js');
var google_api_annotations_pb = require('../google/api/annotations_pb.js');
var google_protobuf_empty_pb = require('google-protobuf/google/protobuf/empty_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');

function serialize_api_CreateDeviceProfileRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.CreateDeviceProfileRequest)) {
    throw new Error('Expected argument of type api.CreateDeviceProfileRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_CreateDeviceProfileRequest(buffer_arg) {
  return api_device_profile_pb.CreateDeviceProfileRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_CreateDeviceProfileResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.CreateDeviceProfileResponse)) {
    throw new Error('Expected argument of type api.CreateDeviceProfileResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_CreateDeviceProfileResponse(buffer_arg) {
  return api_device_profile_pb.CreateDeviceProfileResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_DeleteDeviceProfileDeviceRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.DeleteDeviceProfileDeviceRequest)) {
    throw new Error('Expected argument of type api.DeleteDeviceProfileDeviceRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_DeleteDeviceProfileDeviceRequest(buffer_arg) {
  return api_device_profile_pb.DeleteDeviceProfileDeviceRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_DeleteDeviceProfileRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.DeleteDeviceProfileRequest)) {
    throw new Error('Expected argument of type api.DeleteDeviceProfileRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_DeleteDeviceProfileRequest(buffer_arg) {
  return api_device_profile_pb.DeleteDeviceProfileRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_DeleteDeviceProfileVendorRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.DeleteDeviceProfileVendorRequest)) {
    throw new Error('Expected argument of type api.DeleteDeviceProfileVendorRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_DeleteDeviceProfileVendorRequest(buffer_arg) {
  return api_device_profile_pb.DeleteDeviceProfileVendorRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileByProfileIdRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileByProfileIdRequest)) {
    throw new Error('Expected argument of type api.GetDeviceProfileByProfileIdRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileByProfileIdRequest(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileByProfileIdRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileByProfileIdResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileByProfileIdResponse)) {
    throw new Error('Expected argument of type api.GetDeviceProfileByProfileIdResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileByProfileIdResponse(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileByProfileIdResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileDeviceRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileDeviceRequest)) {
    throw new Error('Expected argument of type api.GetDeviceProfileDeviceRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileDeviceRequest(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileDeviceRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileDeviceResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileDeviceResponse)) {
    throw new Error('Expected argument of type api.GetDeviceProfileDeviceResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileDeviceResponse(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileDeviceResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileRequest)) {
    throw new Error('Expected argument of type api.GetDeviceProfileRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileRequest(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileResponse)) {
    throw new Error('Expected argument of type api.GetDeviceProfileResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileResponse(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileVendorRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileVendorRequest)) {
    throw new Error('Expected argument of type api.GetDeviceProfileVendorRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileVendorRequest(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileVendorRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_GetDeviceProfileVendorResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.GetDeviceProfileVendorResponse)) {
    throw new Error('Expected argument of type api.GetDeviceProfileVendorResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_GetDeviceProfileVendorResponse(buffer_arg) {
  return api_device_profile_pb.GetDeviceProfileVendorResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfileAdrAlgorithmsResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfileAdrAlgorithmsResponse)) {
    throw new Error('Expected argument of type api.ListDeviceProfileAdrAlgorithmsResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfileAdrAlgorithmsResponse(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfileAdrAlgorithmsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfileDevicesRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfileDevicesRequest)) {
    throw new Error('Expected argument of type api.ListDeviceProfileDevicesRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfileDevicesRequest(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfileDevicesRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfileDevicesResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfileDevicesResponse)) {
    throw new Error('Expected argument of type api.ListDeviceProfileDevicesResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfileDevicesResponse(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfileDevicesResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfileVendorsRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfileVendorsRequest)) {
    throw new Error('Expected argument of type api.ListDeviceProfileVendorsRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfileVendorsRequest(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfileVendorsRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfileVendorsResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfileVendorsResponse)) {
    throw new Error('Expected argument of type api.ListDeviceProfileVendorsResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfileVendorsResponse(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfileVendorsResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfilesRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfilesRequest)) {
    throw new Error('Expected argument of type api.ListDeviceProfilesRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfilesRequest(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfilesRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_ListDeviceProfilesResponse(arg) {
  if (!(arg instanceof api_device_profile_pb.ListDeviceProfilesResponse)) {
    throw new Error('Expected argument of type api.ListDeviceProfilesResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_ListDeviceProfilesResponse(buffer_arg) {
  return api_device_profile_pb.ListDeviceProfilesResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_api_UpdateDeviceProfileRequest(arg) {
  if (!(arg instanceof api_device_profile_pb.UpdateDeviceProfileRequest)) {
    throw new Error('Expected argument of type api.UpdateDeviceProfileRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_api_UpdateDeviceProfileRequest(buffer_arg) {
  return api_device_profile_pb.UpdateDeviceProfileRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_google_protobuf_Empty(arg) {
  if (!(arg instanceof google_protobuf_empty_pb.Empty)) {
    throw new Error('Expected argument of type google.protobuf.Empty');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_google_protobuf_Empty(buffer_arg) {
  return google_protobuf_empty_pb.Empty.deserializeBinary(new Uint8Array(buffer_arg));
}


// DeviceProfileService is the service providing API methods for managing
// device-profiles.
var DeviceProfileServiceService = exports.DeviceProfileServiceService = {
  // Create the given device-profile.
create: {
    path: '/api.DeviceProfileService/Create',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.CreateDeviceProfileRequest,
    responseType: api_device_profile_pb.CreateDeviceProfileResponse,
    requestSerialize: serialize_api_CreateDeviceProfileRequest,
    requestDeserialize: deserialize_api_CreateDeviceProfileRequest,
    responseSerialize: serialize_api_CreateDeviceProfileResponse,
    responseDeserialize: deserialize_api_CreateDeviceProfileResponse,
  },
  // Get the device-profile for the given ID.
get: {
    path: '/api.DeviceProfileService/Get',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.GetDeviceProfileRequest,
    responseType: api_device_profile_pb.GetDeviceProfileResponse,
    requestSerialize: serialize_api_GetDeviceProfileRequest,
    requestDeserialize: deserialize_api_GetDeviceProfileRequest,
    responseSerialize: serialize_api_GetDeviceProfileResponse,
    responseDeserialize: deserialize_api_GetDeviceProfileResponse,
  },
  // GetByProfileId tries to get the device-profile for the given ProfileID (see TR0005).
getByProfileId: {
    path: '/api.DeviceProfileService/GetByProfileId',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.GetDeviceProfileByProfileIdRequest,
    responseType: api_device_profile_pb.GetDeviceProfileByProfileIdResponse,
    requestSerialize: serialize_api_GetDeviceProfileByProfileIdRequest,
    requestDeserialize: deserialize_api_GetDeviceProfileByProfileIdRequest,
    responseSerialize: serialize_api_GetDeviceProfileByProfileIdResponse,
    responseDeserialize: deserialize_api_GetDeviceProfileByProfileIdResponse,
  },
  // Update the given device-profile.
update: {
    path: '/api.DeviceProfileService/Update',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.UpdateDeviceProfileRequest,
    responseType: google_protobuf_empty_pb.Empty,
    requestSerialize: serialize_api_UpdateDeviceProfileRequest,
    requestDeserialize: deserialize_api_UpdateDeviceProfileRequest,
    responseSerialize: serialize_google_protobuf_Empty,
    responseDeserialize: deserialize_google_protobuf_Empty,
  },
  // Delete the device-profile with the given ID.
delete: {
    path: '/api.DeviceProfileService/Delete',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.DeleteDeviceProfileRequest,
    responseType: google_protobuf_empty_pb.Empty,
    requestSerialize: serialize_api_DeleteDeviceProfileRequest,
    requestDeserialize: deserialize_api_DeleteDeviceProfileRequest,
    responseSerialize: serialize_google_protobuf_Empty,
    responseDeserialize: deserialize_google_protobuf_Empty,
  },
  // List the available device-profiles.
list: {
    path: '/api.DeviceProfileService/List',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.ListDeviceProfilesRequest,
    responseType: api_device_profile_pb.ListDeviceProfilesResponse,
    requestSerialize: serialize_api_ListDeviceProfilesRequest,
    requestDeserialize: deserialize_api_ListDeviceProfilesRequest,
    responseSerialize: serialize_api_ListDeviceProfilesResponse,
    responseDeserialize: deserialize_api_ListDeviceProfilesResponse,
  },
  // List the available device-profile vendors.
listVendors: {
    path: '/api.DeviceProfileService/ListVendors',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.ListDeviceProfileVendorsRequest,
    responseType: api_device_profile_pb.ListDeviceProfileVendorsResponse,
    requestSerialize: serialize_api_ListDeviceProfileVendorsRequest,
    requestDeserialize: deserialize_api_ListDeviceProfileVendorsRequest,
    responseSerialize: serialize_api_ListDeviceProfileVendorsResponse,
    responseDeserialize: deserialize_api_ListDeviceProfileVendorsResponse,
  },
  // Get the device-profile vendor.
getVendor: {
    path: '/api.DeviceProfileService/GetVendor',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.GetDeviceProfileVendorRequest,
    responseType: api_device_profile_pb.GetDeviceProfileVendorResponse,
    requestSerialize: serialize_api_GetDeviceProfileVendorRequest,
    requestDeserialize: deserialize_api_GetDeviceProfileVendorRequest,
    responseSerialize: serialize_api_GetDeviceProfileVendorResponse,
    responseDeserialize: deserialize_api_GetDeviceProfileVendorResponse,
  },
  // Delete the given device-profile vendor.
deleteVendor: {
    path: '/api.DeviceProfileService/DeleteVendor',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.DeleteDeviceProfileVendorRequest,
    responseType: google_protobuf_empty_pb.Empty,
    requestSerialize: serialize_api_DeleteDeviceProfileVendorRequest,
    requestDeserialize: deserialize_api_DeleteDeviceProfileVendorRequest,
    responseSerialize: serialize_google_protobuf_Empty,
    responseDeserialize: deserialize_google_protobuf_Empty,
  },
  // List the available device-profile devices.
listDevices: {
    path: '/api.DeviceProfileService/ListDevices',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.ListDeviceProfileDevicesRequest,
    responseType: api_device_profile_pb.ListDeviceProfileDevicesResponse,
    requestSerialize: serialize_api_ListDeviceProfileDevicesRequest,
    requestDeserialize: deserialize_api_ListDeviceProfileDevicesRequest,
    responseSerialize: serialize_api_ListDeviceProfileDevicesResponse,
    responseDeserialize: deserialize_api_ListDeviceProfileDevicesResponse,
  },
  // Get the device-profile device.
getDevice: {
    path: '/api.DeviceProfileService/GetDevice',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.GetDeviceProfileDeviceRequest,
    responseType: api_device_profile_pb.GetDeviceProfileDeviceResponse,
    requestSerialize: serialize_api_GetDeviceProfileDeviceRequest,
    requestDeserialize: deserialize_api_GetDeviceProfileDeviceRequest,
    responseSerialize: serialize_api_GetDeviceProfileDeviceResponse,
    responseDeserialize: deserialize_api_GetDeviceProfileDeviceResponse,
  },
  // Delete the given device-profile device.
deleteDevice: {
    path: '/api.DeviceProfileService/DeleteDevice',
    requestStream: false,
    responseStream: false,
    requestType: api_device_profile_pb.DeleteDeviceProfileDeviceRequest,
    responseType: google_protobuf_empty_pb.Empty,
    requestSerialize: serialize_api_DeleteDeviceProfileDeviceRequest,
    requestDeserialize: deserialize_api_DeleteDeviceProfileDeviceRequest,
    responseSerialize: serialize_google_protobuf_Empty,
    responseDeserialize: deserialize_google_protobuf_Empty,
  },
  // List available ADR algorithms.
listAdrAlgorithms: {
    path: '/api.DeviceProfileService/ListAdrAlgorithms',
    requestStream: false,
    responseStream: false,
    requestType: google_protobuf_empty_pb.Empty,
    responseType: api_device_profile_pb.ListDeviceProfileAdrAlgorithmsResponse,
    requestSerialize: serialize_google_protobuf_Empty,
    requestDeserialize: deserialize_google_protobuf_Empty,
    responseSerialize: serialize_api_ListDeviceProfileAdrAlgorithmsResponse,
    responseDeserialize: deserialize_api_ListDeviceProfileAdrAlgorithmsResponse,
  },
};

exports.DeviceProfileServiceClient = grpc.makeGenericClientConstructor(DeviceProfileServiceService, 'DeviceProfileService');
